loyalty_mass_mailing/models/loyalty_program.py

```diff
@@ -17,6 +18,24 @@ class SaleCouponProgram(models.Model):
         compute="_compute_mailing_count", string="Mailing count"
     )
 
+    partner_applicability_domain = fields.Char(
+        compute="_compute_partner_applicability_domain",
+    )
+
+    @api.depends("rule_ids.rule_partners_domain")
+    def _compute_partner_applicability_domain(self):
+        for program in self:
+            domain_list = self.rule_ids.mapped("rule_partners_domain")
+            if any(domain.strip("[]") == "" for domain in domain_list):
+                program.partner_applicability_domain = "[]"
+            else:
+                program.partner_applicability_domain = (
+                    "["
+                    + "'|'," * (len(domain_list) - 1)
+                    + ",".join(x[1:-1] for x in domain_list)
+                    + "]"
+                )
```

Comment:
Simpler:
```suggestion
            domain_list = [literal_eval(d) for d in self.rule_ids.mapped("rule_partners_domain") if d]
            program.partner_applicability_domain = str(expression.OR([d for d in domain_list]))
```

Reply:
Do you need another comprehension list in the second line?

Reply:
Oh, yeah, it's useless :smile:

Reply:
done!

Reply:
> Do you need another comprehension list in the second line?

I don't think so...